kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala

```diff
@@ -66,8 +66,22 @@ object KernelCheck {
   }
 
   // Copied from cats-laws.
-  implicit def arbitrarySortedSet[A: Arbitrary: Order]: Arbitrary[SortedSet[A]] =
-    Arbitrary(arbitrary[Set[A]].map(s => SortedSet.empty[A](implicitly[Order[A]].toOrdering) ++ s))
+  implicit def arbitrarySortedSet[A: Arbitrary: Order: Cogen]: Arbitrary[SortedSet[A]] = {
+    // We create an arbitrary Ordering[A] which is either the same as Order[A]
+    // or the reverse of it. This is important because the Ordering in use by
+    // a given SortedSet[A] may not be derived from our Order and
+    // scala.math.Ordering does not imply coherence.
+    val orderingGen: Gen[Ordering[A]] =
+      Arbitrary.arbitrary[A => Int].map(order =>
+        new Ordering[A] {
+          override def compare(x: A, y: A): Int =
+            order(x).compare(order(y))
+        }
+      )
+    Arbitrary(
+      orderingGen.flatMap(ordering => arbitrary[Set[A]].map(s => SortedSet.empty[A](ordering) ++ s))
```

Comment:
nit: I think `for` would be prettier here:
```scala
Arbitrary(for {
  ordering <- orderingGen
  setA <- arbitrary[Set[A]]
} yield SortedSet.empty(ordering) ++ setA)
```

Reply:
I'll update that